src/basic/stat-util.c

```diff
@@ -471,3 +471,154 @@ int statx_fallback(int dfd, const char *path, int flags, unsigned mask, struct s
+                *ret = JFFS2_SUPER_MAGIC;
+        else if (streq(name, "minix"))
+                *ret = MINIX3_SUPER_MAGIC;
+        else if (streq(name, "mqueue"))
+                *ret = MQUEUE_MAGIC;
+        else if (streq(name, "msdos"))
+                *ret = MSDOS_SUPER_MAGIC;
+        else if (streq(name, "ncpfs"))
+                *ret = NCP_SUPER_MAGIC;
+        else if (streq(name, "nfs"))
+                *ret = NFS_SUPER_MAGIC;
+        else if (streq(name, "nilfs2"))
+                *ret = NILFS_SUPER_MAGIC;
+        else if (streq(name, "nsfs"))
+                *ret = NSFS_MAGIC;
+        else if (streq(name, "ocfs2"))
+                *ret = OCFS2_SUPER_MAGIC;
+        else if (streq(name, "openpromfs"))
+                *ret = OPENPROM_SUPER_MAGIC;
+        else if (streq(name, "overlay"))
+                *ret = OVERLAYFS_SUPER_MAGIC;
+        else if (streq(name, "pipefs"))
+                *ret = PIPEFS_MAGIC;
+        else if (streq(name, "ppc-cmm"))
+                *ret = PPC_CMM_MAGIC;
+        else if (streq(name, "proc"))
+                *ret = PROC_SUPER_MAGIC;
+        else if (streq(name, "pstore"))
+                *ret = PSTOREFS_MAGIC;
+        else if (streq(name, "qnx4"))
+                *ret = QNX4_SUPER_MAGIC;
+        else if (streq(name, "qnx6"))
+                *ret = QNX6_SUPER_MAGIC;
+        else if (streq(name, "ramfs"))
+                *ret = RAMFS_MAGIC;
+        else if (streq(name, "resctrl"))
+                *ret = RDTGROUP_SUPER_MAGIC;
+        else if (streq(name, "reiserfs"))
+                *ret = REISERFS_SUPER_MAGIC;
+        else if (streq(name, "securityfs"))
+                *ret = SECURITYFS_MAGIC;
+        else if (streq(name, "selinuxfs"))
+                *ret = SELINUX_MAGIC;
+        else if (streq(name, "smackfs"))
+                *ret = SMACK_MAGIC;
+        else if (streq(name, "smb3"))
+                *ret = SMB_SUPER_MAGIC;
+        else if (streq(name, "sockfs"))
+                *ret = SOCKFS_MAGIC;
+        else if (streq(name, "squashfs"))
+                *ret = SQUASHFS_MAGIC;
+        else if (streq(name, "sysfs"))
+                *ret = SYSFS_MAGIC;
+        else if (streq(name, "tmpfs"))
+                *ret = TMPFS_MAGIC;
+        else if (streq(name, "tracefs"))
+                *ret = TRACEFS_MAGIC;
+        else if (streq(name, "udf"))
+                *ret = UDF_SUPER_MAGIC;
+        else if (streq(name, "usbdevfs"))
+                *ret = USBDEVICE_SUPER_MAGIC;
+        else if (streq(name, "v9fs"))
+                *ret = V9FS_MAGIC;
+        else if (streq(name, "xenfs"))
+                *ret = XENFS_SUPER_MAGIC;
+        else if (streq(name, "xfs"))
+                *ret = XFS_SUPER_MAGIC;
+        else if (streq(name, "z3fold"))
+                *ret = Z3FOLD_MAGIC;
+        else if (streq(name, "zonefs"))
+                *ret = ZONEFS_MAGIC;
+        else if (streq(name, "zsmalloc"))
+                *ret = ZSMALLOC_MAGIC;
```

Comment:
How would a user find that list of magic names ? is that the same one as the one the "mount" comand use ?
how to make that future-proof wrt futre filesystems or out-of-tree filesystems ? is there a better way to do that if/elsif list ?

Reply:
This is a good point, I've used the names in the `file_system_type.name` struct in the kernel, I assume they're the same ones the "mount" command uses but I'm not sure.

I don't have good ideas on how to make it future proof, I don't think there's any API to get the magic number for a name so it'd have to be something manual.